<!DOCTYPE html>
<html lang=" en-US">

<head>
  <link rel="preconnect" href="https://fonts.googleapis.com">
  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
  <link href="https://fonts.googleapis.com/css2?family=Cormorant+Garamond:ital,wght@0,300;0,400;0,500;0,600;0,700;1,300;1,400;1,500;1,600;1,700&display=swap" rel="stylesheet">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta charset="utf-8">
  <title>Andrew Lynch</title>
  <link rel="stylesheet" href="/assets/css/main.css">
</head>
<body>
  <div class="site-container">
    <div class="site-header-container grid-width-full">
      <a href = "http://localhost:4000"><b>Andrew Lynch</a></b> &rarr; Posts (reel)
    </div>
    <div class="site-body-container grid-width-full">
      <p><a href="/posts">Index</a></p>

<h1><a href="/posts/2022-02-14-thoughts-on-fitness-in-karyotype-space">Thoughts on visualizing fitness in karyotype space</a></h1>
<p><b>14 Feb, 2022</b></p>
    <p>The publication of our <a href="https://www.biorxiv.org/content/10.1101/2021.04.26.441466v1">pre-print</a> on advancing quantification of chromosomal instability is nigh. I’m wrapped up a long stretch of revisions for the re-submit and, as a result, have been thinking a great deal about karyotype selection. Ultimately, we want to be able to experimentally quantify selection. We’re not there yet, but how would we represent this when that time comes? Reviews on the effects of selection, in general, are rife with toy visualizations of ‘genotype spaces’. The <em>concept</em> is understandable to me and is easily generalized to the biology of aneuploidy — a ‘karyotype space’. However, I’ve recently been attempting to plot out a fitness landscape for karyotypes and it’s clear I underestimated the difficulty of implementing a <em>useful</em> visualization of karyotypic space.<br />
<img src="https://upload.wikimedia.org/wikipedia/commons/e/ea/Visualization_of_two_dimensions_of_a_NK_fitness_landscape.png" class="img-center img-66" /></p>

<p>I want to be clear about what I’m trying to produce. I want a granular landscape where fitness values are defined for every unique karyotype from haploid to tetraploid presented in an intuitive and useful way.</p>
<h3 id="permutation-through-karyotypic-space">Permutation through karyotypic space</h3>
<p>My first impressions of fitness landscapes were conceptual. In reality, there are 4<sup>23</sup> unique karyotypes between an <em>n</em> of 1 and 4 and I don’t think it would necessarily be useful to consider all 70 trillion+. Let’s consider a smaller karyotypic space of, say, 8 chromosomes. Permuting yields 65,536 unique karyotypes.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">arrangements</span><span class="p">)</span><span class="w">
</span><span class="c1">#Selecting copy numbers 1-4 for k chromosomes</span><span class="w">
</span><span class="n">k</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">8</span><span class="w">
</span><span class="n">karyos</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.data.frame</span><span class="p">(</span><span class="n">permutations</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">,</span><span class="n">k</span><span class="o">=</span><span class="n">k</span><span class="p">,</span><span class="n">replace</span><span class="o">=</span><span class="nb">T</span><span class="p">))</span><span class="w">
</span></code></pre></div></div>

<h3 id="calculating-ploidy-aneuploidy-and-theoretical-fitness-of-each-karyotype">Calculating ploidy, aneuploidy, and <em>theoretical</em> fitness of each karyotype</h3>
<p>We can now calculate the individual characteristics of these karyotypes that are actually interesting. Ploidy and aneuploidy are pretty easy to calculate as just the average copy number and average intra-karyotype variance respectively. But how do you calculate fitness of aneuploid karyotypes? One way to look at this is through the lens of stabilizing selection—selection against clones with karyotypes that stray too far from a ‘core’ karyotype. Current evidence, at least in non-cancer-derived tissue suggests that aneuploid cellular fitness is negatively correlated to the number of genes on the chromosome(s) for which a cell is aneuploid<sup>1,2</sup>. For example, aneuploidy of chromosome 1 having a larger fitness detriment than that of chromosome 18 [1,2]. So in this case it’s all about balancing the stoichiometry of expressed genes.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Sum the number of annotated genes on each chromosome</span><span class="w">
</span><span class="n">chrscores</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">2880</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">2615</span><span class="p">,</span><span class="w"> </span><span class="m">1647</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">2557</span><span class="p">,</span><span class="w"> </span><span class="m">1392</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1806</span><span class="p">,</span><span class="w"> </span><span class="m">746</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1917</span><span class="p">,</span><span class="w"> </span><span class="m">710</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">2282</span><span class="p">,</span><span class="w"> </span><span class="m">1541</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1539</span><span class="p">,</span><span class="w"> </span><span class="m">1082</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1947</span><span class="p">,</span><span class="w"> </span><span class="m">959</span><span class="w"> </span><span class="o">+</span><span class="w"> </span><span class="m">1536</span><span class="p">)</span><span class="w">

</span><span class="c1"># Normalize the gene abundance of each chromosome to the sum across the karyotype to get relative gene abundance (relative importance to fitness). </span><span class="w">
</span><span class="n">chrscores</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">chrscores</span><span class="w"> </span><span class="o">/</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">chrscores</span><span class="w">

</span><span class="c1"># Calculate the ploidy for each karyotype</span><span class="w">
</span><span class="n">karyos</span><span class="o">$</span><span class="n">ploidy</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">apply</span><span class="p">(</span><span class="n">karyos</span><span class="p">[,</span><span class="m">1</span><span class="o">:</span><span class="n">k</span><span class="p">],</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span><span class="nf">sum</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="o">/</span><span class="n">k</span><span class="p">})</span><span class="w">

</span><span class="c1"># Calculate the aneuploidy of each karyotype</span><span class="w">
</span><span class="n">karyos</span><span class="o">$</span><span class="n">aneuploidy</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">apply</span><span class="p">(</span><span class="n">karyos</span><span class="p">[,</span><span class="m">1</span><span class="o">:</span><span class="n">k</span><span class="p">],</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span><span class="n">var</span><span class="p">(</span><span class="n">x</span><span class="p">)})</span><span class="w">

</span><span class="c1"># Calculate the fitness of each karyotype (this takes a few seconds)</span><span class="w">
</span><span class="n">karyos</span><span class="o">$</span><span class="n">fitness</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">apply</span><span class="p">(</span><span class="n">karyos</span><span class="p">[,</span><span class="m">1</span><span class="o">:</span><span class="n">k</span><span class="p">],</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span><span class="w">
  </span><span class="n">f</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">vector</span><span class="p">()</span><span class="w"> </span><span class="c1">#Create a vector to hold the modified fitness values of each chromosome</span><span class="w">
  </span><span class="k">for</span><span class="p">(</span><span class="n">c</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="n">k</span><span class="p">){</span><span class="w">
    </span><span class="n">f</span><span class="p">[</span><span class="n">c</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">chrscores</span><span class="p">[</span><span class="n">c</span><span class="p">]</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="p">(</span><span class="n">chrscores</span><span class="p">[</span><span class="n">c</span><span class="p">]</span><span class="w"> </span><span class="o">*</span><span class="w"> </span><span class="p">(</span><span class="nf">abs</span><span class="p">(</span><span class="n">x</span><span class="p">[</span><span class="n">c</span><span class="p">]</span><span class="w"> </span><span class="o">-</span><span class="w"> </span><span class="n">mean</span><span class="p">(</span><span class="n">x</span><span class="p">))))</span><span class="o">/</span><span class="w"> </span><span class="n">mean</span><span class="p">(</span><span class="n">x</span><span class="p">)</span><span class="w"> </span><span class="c1"># Modify the fitness value of each chromosome by how unbalanced it is from the ploidy of the cell.</span><span class="w">
  </span><span class="p">}</span><span class="w">
  </span><span class="nf">sum</span><span class="p">(</span><span class="n">f</span><span class="p">)</span><span class="w">
</span><span class="p">})</span><span class="w">
</span></code></pre></div></div>

<p>Great! What did we just do? We generated chromosome scores for chromosomes 1-8; we calculated the ploidy and aneuploidy of each karyotype in the set; and we used these values to calculate the fitness of the karyotype by assuming a balanced (i.e. euploid) karyotype has a perfect fitness with all chromosome scores summing to 1. We also modified the chromosome scores such that chromosomes that were more unbalanced from the ploidy of the cell experienced a fitness detriment, so the sum of chromosome scores for that karyotype would be &lt; 1. There are some caveats to these assumptions like 1) the ploidy of 2.5 indicates the cell is already aneuploid so this may not be the best denominator and 2) the assumption that gains will be roughly equally detrimental as losses, which may not be true in normal tissue due to haploinsufficiency of many genes. Yet I think this is a good place to start.</p>

<p>Let’s plot!</p>

<h3 id="two-dimensional-fitness-landscape">Two-dimensional fitness landscape</h3>
<p>Starting with something super simple. Let’s just consider the landscape of all 8-chromosome karyotypes ordered by fitness.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span><span class="w">

</span><span class="n">karyos</span><span class="o">$</span><span class="n">id</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">rownames</span><span class="p">(</span><span class="n">karyos</span><span class="p">)</span><span class="w"> </span><span class="c1"># Assign an id to each karyotype to easily re-order them by fitness</span><span class="w">

</span><span class="n">ggplot</span><span class="p">(</span><span class="n">karyos</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">reorder.factor</span><span class="p">(</span><span class="n">id</span><span class="p">,</span><span class="n">fitness</span><span class="p">),</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="n">fitness</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> 
  </span><span class="n">geom_point</span><span class="p">(</span><span class="n">aes</span><span class="p">(</span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">fitness</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> 
  </span><span class="n">theme_pubready</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> 
  </span><span class="n">theme</span><span class="p">(</span><span class="w">
    </span><span class="n">axis.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
    </span><span class="n">axis.text.y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">9</span><span class="p">),</span><span class="w">
    </span><span class="n">axis.ticks.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
    </span><span class="n">legend.position</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"none"</span><span class="p">,</span><span class="w">
    </span><span class="n">axis.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">10</span><span class="p">)</span><span class="w">
  </span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> 
  </span><span class="n">scale_x_discrete</span><span class="p">(</span><span class="n">expand</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0.1</span><span class="p">,</span><span class="m">0.1</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> 
  </span><span class="n">scale_color_gradient</span><span class="p">(</span><span class="n">low</span><span class="o">=</span><span class="s2">"tomato1"</span><span class="p">,</span><span class="w"> </span><span class="n">high</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"steelblue1"</span><span class="p">,</span><span class="w"> </span><span class="n">limits</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">1</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> 
  </span><span class="n">labs</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="s2">"Karyotype"</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="o">=</span><span class="s2">"Fitness"</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p><img src="/assets/posts/2022-02-14-thoughts-on-fitness-in-karyotype-space/linear.png" width="300" height="300" class="img-center img-66" /></p>

<p>Okay cool! So what does this tell us? In general, karyotypes that are more aneuploid are less fit! That jives with what we know about the biology of aneuploidy in normal, non-cancerous tissue. But it doesn’t appear to be a clean gradient. Looking closely there are interspersed pockets of blue and red, so this doesn’t give us all the information we need to resolve the full fitness landscape.</p>

<h3 id="three-dimensional-fitness-landscapes">Three-dimensional fitness landscapes</h3>
<p>Now let’s shoot for something more like the ‘classic’ fitness landscape — a nice surface plot. Since we already have our x, y, and z (ploidy, aneuploidy, and fitness), we can just go straight to interpolation.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">akima</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">lattice</span><span class="p">)</span><span class="w">

</span><span class="c1"># Spline interpolation, without extrapolation, and using the median of duplicate datapoints. </span><span class="w">
</span><span class="n">interp_fitness</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">interp</span><span class="p">(</span><span class="n">karyos</span><span class="o">$</span><span class="n">ploidy</span><span class="p">,</span><span class="n">karyos</span><span class="o">$</span><span class="n">aneuploidy</span><span class="p">,</span><span class="n">karyos</span><span class="o">$</span><span class="n">fitness</span><span class="p">,</span><span class="n">linear</span><span class="o">=</span><span class="nb">F</span><span class="p">,</span><span class="n">extrap</span><span class="o">=</span><span class="nb">F</span><span class="p">,</span><span class="w"> </span><span class="n">duplicate</span><span class="o">=</span><span class="s2">"median"</span><span class="p">)</span><span class="w">

</span><span class="n">wireframe</span><span class="p">(</span><span class="n">interp_fitness</span><span class="o">$</span><span class="n">z</span><span class="p">,</span><span class="w">
          </span><span class="n">scales</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="n">z</span><span class="o">=</span><span class="nf">list</span><span class="p">(</span><span class="n">distance</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">),</span><span class="w"> </span><span class="n">arrows</span><span class="o">=</span><span class="nb">T</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"black"</span><span class="p">),</span><span class="w">
          </span><span class="n">drape</span><span class="o">=</span><span class="nb">T</span><span class="p">,</span><span class="w">
          </span><span class="n">colorkey</span><span class="o">=</span><span class="nb">F</span><span class="p">,</span><span class="w"> 
          </span><span class="n">col</span><span class="o">=</span><span class="s1">'white'</span><span class="p">,</span><span class="w">
          </span><span class="n">lwd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.2</span><span class="p">,</span><span class="w">
          </span><span class="n">col.regions</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">colorRampPalette</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="s2">"tomato1"</span><span class="p">,</span><span class="w">  </span><span class="s2">"steelblue1"</span><span class="p">))(</span><span class="m">100</span><span class="p">),</span><span class="w"> 
          </span><span class="n">screen</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">z</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">-200</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">-70</span><span class="p">),</span><span class="w">
          </span><span class="n">xlab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Ploidy"</span><span class="p">,</span><span class="w"> 
          </span><span class="n">ylab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Aneuploidy"</span><span class="p">,</span><span class="w"> 
          </span><span class="n">zlab</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="s2">"Fitness"</span><span class="p">,</span><span class="w"> </span><span class="n">rot</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">90</span><span class="p">,</span><span class="w"> </span><span class="n">distance</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span><span class="p">),</span><span class="w">
          </span><span class="n">par.settings</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">axis.line</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">list</span><span class="p">(</span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0</span><span class="p">)))</span><span class="w">
</span></code></pre></div></div>

<p>That’s looking nice! We can see that fitness generally decreases (blue to red) as aneuploidy increases. Also, because non-integer ploidy values generally indicate some degree of aneuploidy, there is a nice ‘wave’ of fitness along the ploidy axis. It also looks like fitness drops off more quickly at lower ploidy values than higher ploidy values. This concurs with the idea that higher ploidy buffers cellular fitness against the negative effects of unbalanced karyotypes. From a stoichiometric standpoint, with a greater denominator, the cells are just less aneuploid. Yet this still doesn’t tell us much about specific karyotypes and how individual copy number alterations interact to alter fitness.</p>

<p><img src="/assets/posts/2022-02-14-thoughts-on-fitness-in-karyotype-space/3d_fitness.png" width="300" height="300" class="img-center img-66" /></p>

<h3 id="n-dimensional-karyotype-space-using-adjacency-networks">n-dimensional karyotype space using adjacency networks</h3>
<p>So to fully understand how single alterations could change fitness, we need to think of karyotype space as a network, visualizing each unique karyotype as a node, and single alterations that link two karyotypes as ridges. First we need some utilities to calculate manhattan distance between karyotypes.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">network</span><span class="p">)</span><span class="w">
</span><span class="c1">#Function to calculate manhattan distance between two vectors</span><span class="w">
</span><span class="n">manhattan_dist</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">a</span><span class="p">,</span><span class="w"> </span><span class="n">b</span><span class="p">){</span><span class="w">
  </span><span class="n">dist</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">abs</span><span class="p">(</span><span class="n">a</span><span class="o">-</span><span class="n">b</span><span class="p">)</span><span class="w">
  </span><span class="n">dist</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="nf">sum</span><span class="p">(</span><span class="n">dist</span><span class="p">)</span><span class="w">
  </span><span class="nf">return</span><span class="p">(</span><span class="n">dist</span><span class="p">)</span><span class="w">
</span><span class="p">}</span><span class="w">

</span><span class="c1">#Function to apply the above to calculate manhattan distances between two karyotypes</span><span class="w">
</span><span class="n">pairwise.manhattan</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span><span class="w">
  </span><span class="n">pairwise.comparisons</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">t</span><span class="p">(</span><span class="n">combn</span><span class="p">(</span><span class="n">nrow</span><span class="p">(</span><span class="n">x</span><span class="p">),</span><span class="m">2</span><span class="p">))</span><span class="w">
  </span><span class="n">results.matrix</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">matrix</span><span class="p">(</span><span class="n">nrow</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">nrow</span><span class="p">(</span><span class="n">x</span><span class="p">),</span><span class="w"> </span><span class="n">ncol</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">nrow</span><span class="p">(</span><span class="n">x</span><span class="p">))</span><span class="w">
  </span><span class="k">for</span><span class="p">(</span><span class="n">i</span><span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="m">1</span><span class="o">:</span><span class="n">nrow</span><span class="p">(</span><span class="n">pairwise.comparisons</span><span class="p">)){</span><span class="w">
    </span><span class="n">vec</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pairwise.comparisons</span><span class="p">[</span><span class="n">i</span><span class="p">,]</span><span class="w">
    </span><span class="n">results.matrix</span><span class="p">[</span><span class="n">vec</span><span class="p">[</span><span class="m">1</span><span class="p">],</span><span class="w"> </span><span class="n">vec</span><span class="p">[</span><span class="m">2</span><span class="p">]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">manhattan_dist</span><span class="p">(</span><span class="n">karyotypes</span><span class="p">[</span><span class="n">vec</span><span class="p">[</span><span class="m">1</span><span class="p">],],</span><span class="w"> </span><span class="n">karyotypes</span><span class="p">[</span><span class="n">vec</span><span class="p">[</span><span class="m">2</span><span class="p">],])</span><span class="w">
    </span><span class="n">results.matrix</span><span class="p">[</span><span class="n">vec</span><span class="p">[</span><span class="m">2</span><span class="p">],</span><span class="w"> </span><span class="n">vec</span><span class="p">[</span><span class="m">1</span><span class="p">]]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">manhattan_dist</span><span class="p">(</span><span class="n">karyotypes</span><span class="p">[</span><span class="n">vec</span><span class="p">[</span><span class="m">1</span><span class="p">],],</span><span class="w"> </span><span class="n">karyotypes</span><span class="p">[</span><span class="n">vec</span><span class="p">[</span><span class="m">2</span><span class="p">],])</span><span class="w">
  </span><span class="p">}</span><span class="w">
  </span><span class="nf">return</span><span class="p">(</span><span class="n">results.matrix</span><span class="p">)</span><span class="w">
</span><span class="p">}</span><span class="w">
</span></code></pre></div></div>

<p>Alright. Now let’s take a look at the “network” starting with a “karyotype” with only 1 chromosome that can have a copy number of 1-4.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">karyo.copies</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="c1">#copies from 1-4</span><span class="w">
</span><span class="n">karyo.set</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="c1">#two chromosomes of a karyotypes</span><span class="w">
</span><span class="n">karyotypes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">permutations</span><span class="p">(</span><span class="n">n</span><span class="o">=</span><span class="n">karyo.copies</span><span class="p">,</span><span class="n">r</span><span class="o">=</span><span class="n">karyo.set</span><span class="p">,</span><span class="n">v</span><span class="o">=</span><span class="m">1</span><span class="o">:</span><span class="n">karyo.copies</span><span class="p">,</span><span class="w"> </span><span class="n">repeats.allowed</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">)</span><span class="w">

</span><span class="c1"># Get the manhattan distance between each karyotype</span><span class="w">
</span><span class="n">pairwise.karyotypes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pairwise.manhattan</span><span class="p">(</span><span class="n">unique</span><span class="p">(</span><span class="n">karyotypes</span><span class="p">))</span><span class="w"> 

</span><span class="c1"># Convert manhattan distances to intersections. All distances not equal to 1 become 0. </span><span class="w">
</span><span class="n">pairwise.karyotypes</span><span class="p">[</span><span class="nf">is.na</span><span class="p">(</span><span class="n">pairwise.karyotypes</span><span class="p">)]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">0</span><span class="w">
</span><span class="n">pairwise.karyotypes</span><span class="p">[</span><span class="n">pairwise.karyotypes</span><span class="w"> </span><span class="o">!=</span><span class="w"> </span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">0</span><span class="w">
</span><span class="n">pairwise.karyotypes.manhattan.1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">pairwise.karyotypes</span><span class="w">
</span><span class="n">diag</span><span class="p">(</span><span class="n">pairwise.karyotypes</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="m">0</span><span class="w">

</span><span class="c1"># Create the network</span><span class="w">
</span><span class="n">net</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">as.network</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">pairwise.karyotypes.manhattan.1</span><span class="p">,</span><span class="w"> </span><span class="c1"># the network object</span><span class="w">
                  </span><span class="n">directed</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="p">,</span><span class="w"> </span><span class="c1"># network direction not necessary</span><span class="w">
                  </span><span class="n">loops</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">F</span><span class="p">,</span><span class="w"> </span><span class="c1"># no self interactions</span><span class="w">
                  </span><span class="n">matrix.type</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"adjacency"</span><span class="w">
</span><span class="p">)</span><span class="w">

</span><span class="c1"># Create an attribute for vertices that represent euploid karyotypes</span><span class="w">
</span><span class="n">set.vertex.attribute</span><span class="p">(</span><span class="n">net</span><span class="p">,</span><span class="w"> 
                     </span><span class="s2">"Euploid"</span><span class="p">,</span><span class="w"> 
                     </span><span class="n">apply</span><span class="p">(</span><span class="n">unique</span><span class="p">(</span><span class="n">karyotypes</span><span class="p">),</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span><span class="nf">length</span><span class="p">(</span><span class="n">unique</span><span class="p">(</span><span class="n">x</span><span class="p">))</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">1</span><span class="p">})</span><span class="w"> 
</span><span class="p">)</span><span class="w">

</span><span class="c1"># Create labels for each vertex that represent the karyotype</span><span class="w">
</span><span class="n">network.vertex.names</span><span class="p">(</span><span class="n">net</span><span class="p">)</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">apply</span><span class="p">(</span><span class="n">karyotypes</span><span class="p">,</span><span class="m">1</span><span class="p">,</span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span><span class="n">paste</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="n">collapse</span><span class="o">=</span><span class="s2">""</span><span class="p">)})</span><span class="w">

</span><span class="c1"># Make euploid karyotypes orange and aneuploid karyotypes grey</span><span class="w">
</span><span class="n">nodecolors</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ifelse</span><span class="p">(</span><span class="n">apply</span><span class="p">(</span><span class="n">unique</span><span class="p">(</span><span class="n">karyotypes</span><span class="p">),</span><span class="w"> </span><span class="m">1</span><span class="p">,</span><span class="w"> </span><span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span><span class="nf">length</span><span class="p">(</span><span class="n">unique</span><span class="p">(</span><span class="n">x</span><span class="p">))</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="m">1</span><span class="p">}),</span><span class="w"> </span><span class="s2">"Orange"</span><span class="p">,</span><span class="w"> </span><span class="s2">"Grey"</span><span class="p">)</span><span class="w">

</span><span class="c1"># Gimme that network plot</span><span class="w">
</span><span class="n">plot.network</span><span class="p">(</span><span class="n">net</span><span class="p">,</span><span class="w">
             </span><span class="n">vertex.col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">nodecolors</span><span class="p">,</span><span class="w">
             </span><span class="n">vertex.cex</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="p">,</span><span class="w">
             </span><span class="n">displaylabels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nb">T</span><span class="p">,</span><span class="w">
             </span><span class="n">label.pos</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">5</span><span class="p">,</span><span class="w">
             </span><span class="n">label.cex</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.6</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p><img src="/assets/posts/2022-02-14-thoughts-on-fitness-in-karyotype-space/networkplots.png" width="auto" class="img-center" /></p>

<p>As you can see things quickly get out of control as dimensionality increases. And the real karyotype space has 23 dimensions! I think the most visually useful plot was the 3D surface, but that is derived under the assumption that always decreases with aneuploidy. While this is probably a fair assumption in non-tumor tissue, we don’t actually know what that surface looks like. I think these networks are a more interesting way to think about the karyotype fitness landscape, but I think more work is needed to figure out how these could be useful visually. In the mean time, perhaps it would be useful to look at specific copy number interactions in this way?</p>

<h3 id="footnotes">Footnotes</h3>
<ol>
  <li><a href="https://www.biorxiv.org/content/biorxiv/early/2021/08/31/2021.08.31.458318.full.pdf">DOI:10.1101/2021.08.31.458318</a></li>
  <li><a href="https://pubmed.ncbi.nlm.nih.gov/21872963/">DOI:10.1016/j.tig.2011.07.003</a></li>
</ol>


<h1><a href="/posts/2021-10-21-theme-pubready">theme_pubready | a ggplot theme for small, publication-ready plots</a></h1>
<p><b>21 Oct, 2021</b></p>
    <p>A while back I found myself creating a lot of small plots where I was removing the gridlines and reducing the size of axis text as a way to ‘declutter’ the small visuals. To save time, I generated this ggplot theme.</p>

<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">theme_pubready</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="k">function</span><span class="p">(){</span><span class="w">
   </span><span class="n">font</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="s2">"Arial"</span><span class="w">
   </span><span class="n">theme_minimal</span><span class="p">()</span><span class="w"> </span><span class="o">%+replace%</span><span class="w">
   
      </span><span class="n">theme</span><span class="p">(</span><span class="w">
         </span><span class="n">panel.grid</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
         </span><span class="n">panel.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_rect</span><span class="p">(</span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"black"</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">),</span><span class="w">
         </span><span class="n">axis.ticks</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_line</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.25</span><span class="p">),</span><span class="w">
         </span><span class="n">axis.line</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
         </span><span class="n">axis.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">6</span><span class="p">),</span><span class="w">
         </span><span class="n">axis.title</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">7</span><span class="p">),</span><span class="w">
         </span><span class="n">strip.background</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_blank</span><span class="p">(),</span><span class="w">
         </span><span class="n">strip.text</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">size</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">7</span><span class="p">)</span><span class="w">
      </span><span class="p">)</span><span class="w">
</span><span class="p">}</span><span class="w">
</span></code></pre></div></div>

<p>Now let’s plot an example.</p>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="n">ggpubr</span><span class="p">)</span><span class="w">

</span><span class="n">df</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">data.frame</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">rnorm</span><span class="p">(</span><span class="m">50</span><span class="p">),</span><span class="n">y</span><span class="o">=</span><span class="n">rnorm</span><span class="p">(</span><span class="m">50</span><span class="p">))</span><span class="w">

</span><span class="n">p1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ggplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> 
   </span><span class="n">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> 
   </span><span class="n">ggtitle</span><span class="p">(</span><span class="s2">"theme_grey"</span><span class="p">)</span><span class="w">
   
</span><span class="n">p2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ggplot</span><span class="p">(</span><span class="n">df</span><span class="p">,</span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="o">=</span><span class="n">x</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="n">y</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> 
   </span><span class="n">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> 
   </span><span class="n">theme_pubready</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> 
   </span><span class="n">ggtitle</span><span class="p">(</span><span class="s2">"theme_pubready"</span><span class="p">)</span><span class="w">
   
</span><span class="n">ggarrange</span><span class="p">(</span><span class="n">p1</span><span class="p">,</span><span class="n">p2</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>
<p><br />
<img src="/assets/posts/2021-10-21-theme-pubready/theme_pubready_example.png" class="img-center img-66" /></p>


<h1><a href="/posts/2019-12-18-end-of-year-decade-wrap-up">A Decade In-Review</a></h1>
<p><b>31 Dec, 2019</b></p>
    <p>Well here we are. We’re nearing the end of the Twenty-Tens. A decade of increasing global unrest, political polarization, climate change-related weather anomalies, CRISPR-babies, and Taylor Swift!
<br /><br />
This year - 2019 - has been a great year for me. And this decade offers a lot to reflect on - within my own personal life and without. I’d like to kick off a new decade with a ‘mental flush’ of the events that resonated with me as defining the last 10 years.</p>

<p><br /></p>
<h2 id="notable-events">Notable Events</h2>
<h4 id="2010-">2010 <br /></h4>
<p>-The Affordable Care Act is signed into law. <br />
-A 7.0 Mw earthquake devastates the country of Haiti, killing hundreds of thousands.<br />
-The Deepwater Horizon oil rig explodes in the Gulf of Mexico, triggering an environmental crisis.<br /> 
-President Obama repeals “Don’t Ask, Don’t Tell”.<br /></p>
<h4 id="2011">2011<br /></h4>
<p>-World population exceeds 7 billion.<br />
-Osama bin Laden killed by U.S. Special Forces in Pakistan. <br />
-Egyptian revolution begins a series of protests and revolutions in what is referred to as the “Arab Spring”.<br /></p>
<h4 id="2012">2012<br /></h4>
<p>-<a href="https://github.com/andrewrlynch/andrewrlynch.github.io/blob/master/Jinek-et-al.-2012.pdf">CRISPR can be used for <em>targeted</em> DNA cleavage.</a><br />
-<a href="https://www.nytimes.com/2012/07/05/science/cern-physicists-may-have-discovered-higgs-boson-particle.html">Higgs boson is detected at CERN.</a><br />
-The whole Kony 2012 thing. <br />
-The Mayan/Mesoamerican Long Count Calendar ends (21 December). Apocalypse begins. <br /></p>
<h4 id="2013-">2013 <br /></h4>
<p>-Pope Francis succeeds Pope Benedict XVI. <br />
-The Boston Marathon is bombed as runners cross the finish line, killing 3. Another is killed during the subsequent manhunt. <br />
-The passing of Nelson Mandela. <br />
-#BlackLivesMatter rises in the wake of the police-shooting deaths of Trayvon Martin, Michael Brown, and Eric Garner. <br /></p>
<h4 id="2014-">2014 <br /></h4>
<p>-Russia annexes Crimea. <br />
-Ebola epidemic in West Africa. <br />
-The disappearance of Malaysia Airlines Flight 370. <br />
-Barack Obama wears a tan suit and people get upset? <br /></p>
<h4 id="2015-">2015 <br /></h4>
<p>-Officials from 195 countries address climate change intervention in the Paris Climate Agreement. <br />
-Same-sex marriage bans are struck down in all states after hearing in the Supreme Court.  <br />
-Charleston church shooting (and too many others). Mass shootings becoming commonplace. <br /></p>
<h4 id="2016-">2016 <br /></h4>
<p>-Brexit. <br />
-Russia’s interference in U.S. election. <br />
-Donald Trump is elected president. <em>Fuck</em>. <br />
-North Korea scales up nuclear tests. <br />
-The Pulse Nightclub shooting in Orlando, Florida. 50 are killed. <br />
-Gravity waves are detected. <br /></p>
<h4 id="2017-">2017 <br /></h4>
<p>-Donald Trump assumes office. Michael Flynn almost immediately resigns as National Security Director, setting the tone for the presidency. <br />
-Ronan Farrow’s reporting on the accounts of Harvey Weinstein’s sexual misconduct kicks off the #MeToo movement. Allegations against a series of high profile celebrities will follow (Bill Cosby, Kevin Spacey, Louis C.K., Matt Lauer) <br />
-Solar eclipse occurs on my birthday! I observed it from an OfficeMax parking lot. <br />
-58 people killed at a concert in Las Vegas in another mass shooting. This is one of many. <br />
-Comey, Comey, Comey, Comey. <br /></p>
<h4 id="2018-">2018 <br /></h4>
<p>-Trade war begins between U.S. and China. <br />
-Trump administration enacts zero-tolerance family separation policy at the southern border, spearheaded by Senior Policy Advisor and sleepy-eyed White Nationalist, Stephen Miller. <br />
-Democrats win back the House of Representatives. <br />
-More shootings.</p>
<h4 id="2019-">2019 <br /></h4>
<p>-The Mueller Report and the impeachment of Donald Trump. <br />
-World Cup victory for the U.S. Women’s National Soccer Team! <br />
-More shootings. New Zealand enacts ban on military-style weapons.<br />
-First all woman space walk. After an embarrassing NASA space-suit snafu. <br />
-CRISPR-babies are born in China. Scientist He Jiankui sentenced to prison. <br />
-Taylor Swift named Billboard Artist of the Decade. Debate ensues. <br /></p>

<h2 id="for-me---notable-dates-and-milestones-">For Me - Notable Dates and Milestones <br /></h2>
<p><strong>2010</strong> - Truly nothing notable. Playing a lot of World of Warcraft. <br />
<strong>Sometime in 2011</strong> - Realize I should probably get my shit together and start thinking about college. Decide I want to be a psychiatrist. <br />
<strong>31 October 2011</strong> - Meet my future wife in the basement of my girlfriend’s house during a Halloween screening of <em>Cabin Fever</em>. <br />
<strong>11 November 2011</strong> - Wait in line for the midnight release of <em>Elder Scrolls: Skyrim</em>. My best friend and I were there for about four hours before anyone else showed up. <br />
<strong>March? 2012</strong> - Rejected from my hometown college/college of choice. I write an appeal letter. <br />
<strong>April 2012</strong> - Accepted to college of choice! University of Wisconsin - Eau Claire. Go Blugolds! <br />
<strong>June 2012</strong> - Graduate from Chippewa Falls Senior Highschool (Chi-Hi).<br />
<strong>December 2012</strong> - Finish my first semester in college with a 4.0. <strong>Despite this, develop severe imposter syndrome.</strong> Spend a non-zero amount of time reassuring people that the world will not end on 21 Dec 2012. <br />
<strong>December 2013</strong> - Began my cancer research career with Dr. Thao Yang synthesizing peptides and doing 2D NMR. Change my major from Pre-Med Psychology to Pre-Med Biochemistry and Molecular Biology. Start thinking about a career in neurosurgery. <br />
<strong>March 2014</strong> - First big research conference poster presentation. American Chemical Society meeting in Denver Colorado. Paper published in the <em>Journal of Undergraduate Chemistry Research</em><br />
<strong>June-July 2014</strong> - Spend a month in Ecuador for a infectious disease research internship through Ohio University and the Tropical Disease Institute. <br />
<strong>August 2014</strong> - Wisdom teeth removed. Am officially an adult. In my post-surgical stupor, and in very quick succession, take a selfie with my nurse and reveal to my mother that my then-girlfriend and I have broken up.<br />
<strong>September 2015</strong> - Begin dating future-wife, Erin. She’s great. <br />
<strong>June-August 2016</strong> - Serve as peer mentor for a chemistry REU at UWEC. <br />
<strong>June 2016</strong> - Start rock climbing. <br />
<strong>August 2016</strong> - Decide to pursue a PhD instead of going to medical school <br />
<strong>December 2016</strong> - Graduate <em>Cum Laude</em> from UWEC. Wrapping up graduate school applications. <br />
<strong>December 2016-June 2017</strong> - Continuing my college job as a caregiver for survivors of traumatic brain injury. <br />
<strong>July 2017</strong> - Roadtrip with Erin to national parks in the western U.S. <strong>Get engaged.</strong><br />
<strong>August 2017</strong> - Move to Madison, Wisconsin. Pursuing my PhD in UW-Madison’s Cellular and Molecular Pathology training program. Receive first T32.<br />
<strong>November 2018</strong> - Adopt an adorable Ratcha named Sassy. <br />
<strong>January 2018</strong> - Join Dr. Mark Burkard’s laboratory studying chromosomal instability in breast cancer. <br />
<strong>August 2018</strong> - First <strong>big</strong> experiment is a success. <br />
<strong>December 2019</strong> - First national research conference as PhD student at the American Society for Cell Biology meeting in San Diego. <br />
<strong>February 2019</strong> - Recieve second T32 through the UW Genomic Sciences Training Program. <br />
<strong>July 2019</strong> - Get married to Erin. She planned a hell of a party. <br />
<strong>September 2019</strong> - Begin working on first manuscript. Preparing for preliminary exam. <br />
<strong>December 2019</strong> - Genomic Sciences T32 is renewed. Submit NIH F31 application. Still waiting for preliminary exam. Scheduling issues. <br />
<strong>31 December 2019</strong> - Attend gigantic NYE party. We leave at 11:22 PM to ring in the New Year at home with our dog, Sassy. <br /></p>

<h2 id="the-decade-in-data">The Decade in Data</h2>
<h3 id="climate-change">Climate Change</h3>
<div class="language-r highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">#Sea Surface Temperature Anomaly</span><span class="w">
</span><span class="n">p1</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ggplot</span><span class="p">(</span><span class="n">df2</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Year</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Annual.anomaly</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> 
  </span><span class="n">annotate</span><span class="p">(</span><span class="s2">"rect"</span><span class="p">,</span><span class="w"> </span><span class="n">xmin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2010</span><span class="p">,</span><span class="w"> </span><span class="n">xmax</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2020</span><span class="p">,</span><span class="w"> </span><span class="n">ymin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="kc">Inf</span><span class="p">,</span><span class="w"> </span><span class="n">ymax</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">Inf</span><span class="p">,</span><span class="w"> </span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"orange"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_line</span><span class="p">(</span><span class="n">lwd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> 
  </span><span class="n">stat_smooth</span><span class="p">(</span><span class="n">method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"gam"</span><span class="p">,</span><span class="w"> </span><span class="n">formula</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y</span><span class="o">~</span><span class="n">s</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="n">bs</span><span class="o">=</span><span class="s2">"cs"</span><span class="p">),</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"magenta"</span><span class="p">,</span><span class="w"> </span><span class="n">lty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"grey22"</span><span class="p">,</span><span class="w"> </span><span class="n">lwd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1.2</span><span class="p">,</span><span class="w"> </span><span class="n">fullrange</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">se</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> 
  </span><span class="n">labs</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Year"</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Sea Surface Temperature Anomaly (°F)"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> 
  </span><span class="n">xlim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">1958</span><span class="p">,</span><span class="m">2100</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> 
  </span><span class="n">theme_pubclean</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> 
  </span><span class="n">theme</span><span class="p">(</span><span class="n">panel.border</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_rect</span><span class="p">(</span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"black"</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="o">=</span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="o">=</span><span class="m">0.8</span><span class="p">),</span><span class="w">
        </span><span class="n">axis.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">90</span><span class="p">,</span><span class="w"> </span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> 
  </span><span class="n">scale_x_continuous</span><span class="p">(</span><span class="n">breaks</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">seq</span><span class="p">(</span><span class="m">1960</span><span class="p">,</span><span class="m">2100</span><span class="p">,</span><span class="w"> </span><span class="m">10</span><span class="p">),</span><span class="w"> </span><span class="n">limits</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">1958</span><span class="p">,</span><span class="m">2100</span><span class="p">))</span><span class="w">
  
</span><span class="c1">#Atmospheric CO2 Concentration</span><span class="w">
</span><span class="n">p2</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ggplot</span><span class="p">(</span><span class="n">monthly_in_situ_co2_mlo</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">yearmonth</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">seasonally</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> 
  </span><span class="n">annotate</span><span class="p">(</span><span class="s2">"rect"</span><span class="p">,</span><span class="w"> </span><span class="n">xmin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2010</span><span class="p">,</span><span class="w"> </span><span class="n">xmax</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2020</span><span class="p">,</span><span class="w"> </span><span class="n">ymin</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="kc">Inf</span><span class="p">,</span><span class="w"> </span><span class="n">ymax</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">Inf</span><span class="p">,</span><span class="w"> </span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"orange"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> 
  </span><span class="n">geom_line</span><span class="p">(</span><span class="n">lwd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> 
  </span><span class="n">stat_smooth</span><span class="p">(</span><span class="n">method</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"gam"</span><span class="p">,</span><span class="w"> </span><span class="n">formula</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">y</span><span class="o">~</span><span class="n">s</span><span class="p">(</span><span class="n">x</span><span class="p">,</span><span class="n">bs</span><span class="o">=</span><span class="s2">"cs"</span><span class="p">),</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"magenta"</span><span class="p">,</span><span class="w"> </span><span class="n">lty</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">3</span><span class="p">,</span><span class="w"> </span><span class="n">col</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"grey22"</span><span class="p">,</span><span class="w"> </span><span class="n">lwd</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1.2</span><span class="p">,</span><span class="w"> </span><span class="n">fullrange</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">,</span><span class="w"> </span><span class="n">se</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">labs</span><span class="p">(</span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Atmosphere [CO2] (ppm)"</span><span class="p">,</span><span class="w"> </span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"Year"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w"> 
  </span><span class="n">xlim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">1958</span><span class="p">,</span><span class="m">2100</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> 
  </span><span class="n">ylim</span><span class="p">(</span><span class="nf">c</span><span class="p">(</span><span class="m">300</span><span class="p">,</span><span class="m">650</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme_pubclean</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w"> 
  </span><span class="n">theme</span><span class="p">(</span><span class="n">panel.border</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_rect</span><span class="p">(</span><span class="n">colour</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"black"</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="o">=</span><span class="kc">NA</span><span class="p">,</span><span class="w"> </span><span class="n">size</span><span class="o">=</span><span class="m">0.8</span><span class="p">),</span><span class="w">
        </span><span class="n">axis.text.x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_text</span><span class="p">(</span><span class="n">angle</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">90</span><span class="p">,</span><span class="w"> </span><span class="n">hjust</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">1</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w"> 
  </span><span class="n">scale_x_continuous</span><span class="p">(</span><span class="n">breaks</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">seq</span><span class="p">(</span><span class="m">1960</span><span class="p">,</span><span class="m">2100</span><span class="p">,</span><span class="w"> </span><span class="m">10</span><span class="p">),</span><span class="w"> </span><span class="n">limits</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="m">1958</span><span class="p">,</span><span class="m">2100</span><span class="p">))</span><span class="w">

</span><span class="n">ggarrange</span><span class="p">(</span><span class="n">p1</span><span class="p">,</span><span class="w"> </span><span class="n">p2</span><span class="p">)</span><span class="w">
</span></code></pre></div></div>

<p><img src="/assets/posts/2019-12-18-end-of-year-decade-wrap-up/Climate.png" class="img-center" /></p>

<h3 id="google-search-trends">Google Search Trends</h3>
<p><img src="/assets/posts/2019-12-18-end-of-year-decade-wrap-up/Trends.png" class="img-center" /></p>


<h1><a href="/posts/2018-11-20-reflections-redefining-metric-system">Reflections on the redefining of the metric system</a></h1>
<p><b>20 Nov, 2018</b></p>
    <p>On 16 November, at the 26th General Conference on Weights and Measures in Versailles, a vote was cast to overhaul the International System of Units (SI). Historically, the kilogram was based on the mass of an actual physical object, a cylinder of platinum alloy that has been housed in Paris since 1889. This cylinder,  or “Le Grand K”, weighs <em>exactly</em> one kilogram, because it <em>is</em> the kilogram. Unfortunately for Le Grand K and its identical twins, their masses have been found to be divergent, indicating that our definition of the kilogram has been shifting as well. So the question was - in this day and age, is it really acceptable to have transient definitions of our basic units of measurement? How would having units of measurement that are subject to change over time affect us in the near future? What about the far future?</p>

<p>To me, this felt very much like shift from the monetary gold standard to the fiat system we have in place. Money is money because we say it’s money, not because it is worth any amount of gold. Only in this case it’s “a kilogram is a kilogram based on the fundamental relationship between the energy (Planck’s Constant) and mass (E=mc<sup>2</sup> )”. This is the kind of forward thinking that will ensure our success as we develop ever more complex technologies and strike out toward new worlds. Both of which will rely heavily on the universality of our measurements over long periods of time.</p>

<p>The Bureau International des Poids et Mesures even put out a <a href="https://www.youtube.com/watch?v=MUJX1Yyx6TI">cool video</a> about the vote. A little sensational, but it gets the point across.</p>


<h1><a href="/posts/2018-10-21-list-of-old-blog-posts">Old blog posts for which one would ostensibly want a convenient list</a></h1>
<p><b>21 Oct, 2018</b></p>
    <p>Because I’ll be moving my blog from <a href="https://bioconjugate.blog"><em>The Bioconjugate</em></a> to this page, I thought I would leave a trail of breadcrumbs to posts from the old blog for a little while. I’m not sure when of if I’ll shut that space down completely, but most of my online presence will be switching to this space.</p>

<h3 id="august-2018">August 2018</h3>
<p><a href="https://bioconjugate.blog/2018/08/16/science-saturday-feat-the-carbone-cancer-center-at-the-wisconsin-institute-for-discovery/">Science Saturday feat. the Carbone Cancer Center at the Wisconsin Institute for Discovery</a></p>

<h3 id="june-2018">June 2018</h3>
<p><a href="https://bioconjugate.blog/2018/06/26/we-are-living-in-a-fever-dream/">We are living in a fever dream</a></p>

<h3 id="may-2018">May 2018</h3>
<p><a href="https://bioconjugate.blog/2018/05/20/word-of-the-day-tsundoku/">Word of the Day: Tsundoku</a></p>

<h3 id="december-2017">December 2017</h3>
<p><a href="https://bioconjugate.blog/2017/12/08/update-on-blog-and-life/">Update on Blog and Life</a></p>

<h3 id="june-2017">June 2017</h3>
<p><a href="https://bioconjugate.blog/2017/06/10/guest-blog-interplay-of-the-microbiome-and-colorectal-cancer/">Guest Blog - Insterplay of the Microbiome and Colorectal Cancer</a></p>

<h3 id="may-2017">May 2017</h3>
<p><a href="https://bioconjugate.blog/2017/05/31/scientific-communication-in-the-eye-of-the-beholder/">Scientific Communication: In the Eye of the Beholder</a></p>

<p><a href="https://bioconjugate.blog/2017/05/30/climate-change-the-keeling-curve-and-why-you-need-to-be-your-own-scientist/">Climate change, the Keeling Curve, and why you need to be your own scientist</a></p>

<h3 id="march-2017">March 2017</h3>
<p><a href="https://bioconjugate.blog/2017/03/22/why-failure-is-good-for-science/">Why ‘Failure’ is Good for Science</a></p>

<p><a href="https://bioconjugate.blog/2017/03/21/st-pattys-post-beerzymes-alpha-beta-amylase/">St. Patty’s Post: Beerzymes (Alpha &amp; Beta Amylase)</a></p>

<p><a href="https://bioconjugate.blog/2017/02/16/science-of-the-ancients-how-our-distant-predecessors-laid-the-groundwork-for-our-success/">Science of the Ancients: How our Distant Predecessors Laid the Groundwork for our Success</a></p>

<p><a href="https://bioconjugate.blog/2017/02/09/scientific-publisher-pay-walls-an-ethical-nightmare/">Scientific Publisher Pay-Walls: an Ethical Nightmare</a></p>

<h3 id="february-2017">February 2017</h3>
<p><a href="https://bioconjugate.blog/2017/02/07/research-rundown-mouse-study-to-find-link-between-mt-1-protein-and-alzheimers-associated-neural-death/">Research Rundown: Mouse study to find link between MT-1 protein and Alzheimer’s-associated factors</a></p>

<p><a href="https://bioconjugate.blog/2017/02/01/dissecting-a-scientific-journal-article/">Dissecting a Scientific Journal Article</a></p>

<h3 id="january-2017">January 2017</h3>
<p><a href="https://bioconjugate.blog/2017/01/30/scientific-communication-why-is-it-important/">Scientific Communication: Why is it Important?</a></p>



<h1><a href="/posts/2018-03-23-moving">Moving Blog</a></h1>
<p><b>23 Mar, 2018</b></p>
    <p>In an effort to immerse myself in the coding community, I’m moving my preexisting blog at <a href="https://bioconjugate.blog">The Bioconjugate</a> to GitHub.</p>


    </div>
    <div class="site-footer-container grid-width-full">
      <div class="site-footer-content">
        <a href="/about">about</a> ⸱ <a href="https://bsky.app/profile/arlynch.bsky.social">bluesky</a> ⸱ <a href="https://github.com/andrewrlynch">github</a> ⸱ <a href="www.linkedin.com/in/andrewrlynch">linkedin</a> ⸱ <a href="/cv">cv</a> ⸱ <a href="https://orcid.org/0000-0002-0238-682X">orcid</a>
      </div>
    </div>
  </div>
</body>
</html>